File: 04-Arrays/Max.java
package com.Anwesh;

public class Max {
    public static void main(String[] args) {
        int[] arr = {1, 3, 23, 9, 18};
        System.out.println(max(arr));
        System.out.println(maxRange(arr, 1, 4));
    }

    //image that arr is not empty
    static int max(int[] arr) {
        if (arr.length == 0) return -1;
        int max = arr[0];
        for (int i = 0; i < arr.length; i++) {
            if (max < arr[i])
                max = arr[i];
        }
        return max;
    }

    //work on edge cases here, like array being null
    static int maxRange(int[] arr, int start, int end) {
        if (end > start) {
            return -1;
        }
        if (arr == null) return -1;
        int max = arr[start];
        for (int i = start; i <= end; i++) {
            if (arr[i] > max)
                max = arr[i];
        }
        return max;
    }
}

File: 04-Arrays/C/max.c
#include <stdio.h>
#include <stdlib.h>

int max(int* v, int size) {
  int max = v[0];
  for (int i = 0; i < size; i++)
    if (v[i] > max)
      max = v[i];
  return max;
}

int main() {
  int v[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
  printf("%d\n", max(v, 10));
}